### PR TITLE
BFB-135: Reloading UI Feedback

### PR DESCRIPTION
**Pull request by m4rtint**
**JIRA Ticket**
https://perigongames.atlassian.net/browse/BFB-135

**Description**
- Reloading UI updates the reloading duration

**Screenshots / GIFs**
demo1

### CODE REVIEW

**Review by unoctium1 — COMMENTED** [review]
Code looks good, but there are some strange issues on the automatic rifle specifically. Reloading only seems to be working when holding down fire, and if you start reloading by hitting the reload key while bullets are still available, it only gets part of the way through reloading before cancelling the reload and shooting again

Edit: Looking at the code for the AutomaticWeapon, it looks like reloading is behind `if(!_isFiring) return` and its not for semi auto weapons

**Comment by m4rtint** [comment]
> Code looks good, but there are some strange issues on the automatic rifle specifically. Reloading only seems to be working when holding down fire, and if you start reloading by hitting the reload key while bullets are still available, it only gets part of the way through reloading before cancelling the reload and shooting again
> 
> Edit: Looking at the code for the AutomaticWeapon, it looks like reloading is behind `if(!_isFiring) return` and its not for semi auto weapons

ahhhh i see, i'll fix

File: Assets/_BForBoss/_Weapons/Scripts/AutomaticWeaponBehaviour.cs [hunk 1]
```diff
@@ -22,14 +22,13 @@ protected override void OnFireInputAction(InputAction.CallbackContext context)
 
         protected override void Update()
         {
-            if (!_isFiring)
+            if (_isFiring)
             {
-                return;
+                _weapon.DecrementElapsedTimeRateOfFire(Time.deltaTime);
```
Comment (unoctium1) [inline Assets/_BForBoss/_Weapons/Scripts/AutomaticWeaponBehaviour.cs:27]: Shouldn't this also decrement even if not firing?

Comment (m4rtint) [reply Assets/_BForBoss/_Weapons/Scripts/AutomaticWeaponBehaviour.cs]: yup you are correct, updating

**Review by unoctium1 — APPROVED** [review]
Great, lgtm

File: Assets/_BForBoss/_Core/Scripts/ReloadViewBehaviour.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        }
+
+        private void Update()
+        {
+            _reloadView.fillAmount = GetElapsedDurationPercentageDone();
+        }
+
+        private float GetElapsedDurationPercentageDone()
+        {
+            if (_equipmentData != null)
```
Comment (Gursaanj) [inline Assets/_BForBoss/_Core/Scripts/ReloadViewBehaviour.cs:30]: do you want to add a check to make sure that `_equipmentData.MaxReloadDuration` cannot be 0?

**Review by Gursaanj — APPROVED** [review]
Looks good, small comment

**Pull request merged**